packages/itwinui-react/src/core/Table/cells/EditableCell.test.tsx

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { render } from '@testing-library/react';
+import * as React from 'react';
+import { EditableCell } from './EditableCell.js';
+import type { CellRendererProps } from '@itwin/itwinui-react/react-table';
+
+const renderComponent = (
+  props?: Partial<React.ComponentProps<typeof EditableCell>>,
+) => {
+  const dummyTableCellRenderedProps = {
+    cellElementProps: {
+      className: 'iui-table-cell',
+      style: {
+        boxSizing: 'border-box',
+        flex: '1 1 145px',
+        minWidth: '72px',
+        width: '0px',
+      },
+    },
+    cellProps: {
+      column: {
+        id: 'id',
+      },
+      value: 'cell value',
+    },
+  } as CellRendererProps<any>;
+
+  const onCellEdit = vi.fn();
+
+  return render(
+    <EditableCell
+      onCellEdit={onCellEdit}
+      {...dummyTableCellRenderedProps}
+      {...props}
+    />,
+  );
+};
+
+it('should merge className/style with cellElementProps', () => {
+  const { container } = renderComponent({
+    className: 'test-class',
+    style: { color: 'red' },
+  });
+
+  const cellElement = container.querySelector('.iui-table-cell') as HTMLElement;
+  expect(cellElement).toBeTruthy();
+  expect(cellElement.classList.contains('test-class')).toBe(true);
+  expect(cellElement.style.color).toBe('red');
+});
```

Comment:
This can be simplified a lot. Avoid introducing unnecessary abstractions and irrelevant details in unit tests. We only need to test one very specific thing here, so pass only that, along with the minimum _required_ props.

```jsx
it('should merge className/style with cellElementProps', () => {
  const { container } = renderComponent(
    <EditableCell
      cellElementProps={{
        key: 'key',
        className: 'original-class',
        style: { width: 100 },
      }}
      className='test-class'
      style={{ color: 'red' }}
      cellProps={{ value: 'value' } as any}
      onCellEdit={() => {}}
      children={null}
    />
  );

  const cellElement = container.querySelector('.original-class') as HTMLElement;
  expect(cellElement).toHaveClass('test-class');
  expect(cellElement.style.width).toBe('100px');
  expect(cellElement.style.color).toBe('red');
});```

Reply:
Sounds good, removed some unnecessary fields and functions, like your suggested code (https://github.com/iTwin/iTwinUI/pull/2068/commits/528847fa36cb8b50042482c5edafc7c7f56dd76e).